### ISSUE TITLE
Large volume specs prevent container from starting

### ISSUE THREAD

**Issue by steefdebruijn** [docker/for-mac#2968]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:
0720BB9D-AB80-41C1-94BB-4F5E1A512088/20180604-092430

### Expected behavior
Container starts
### Actual behavior
Container start aborts with error.

I have seen two versions of abort:

    $ docker-compose up
    Removing dockervolumebug_thingie3_1
    Starting dockervolumebug_thingie2_1 ... 
    Recreating 9c3da9e15d1f_dockervolumebug_thingie3_1 ... error
    Starting dockervolumebug_thingie2_1                ... done
    Starting dockervolumebug_thingie_1                 ... done

    ERROR: for thingie3  Cannot start service thingie3: b'Mounts denied: write unix ->osxfs.sock: write: broken pipe'
    ERROR: Encountered errors while bringing up the project.

And sometimes I get

    $ docker-compose up
    Creating dockervolumebug_thingie3_1 ... error
    Creating dockervolumebug_thingie2_1 ... 
    Creating dockervolumebug_thingie_1  ... 
    Creating dockervolumebug_thingie2_1 ... done
    Creating dockervolumebug_thingie_1  ... done

    ERROR: for thingie3  Cannot start service thingie3: b'Mounts denied: EOF'
    ERROR: Encountered errors while bringing up the project.

It is not clear to me when I get which error. Behavior changes on if and how the above containers are deleted between runs. (`docker-compose rm` seems to have a different effect than `docker rm xxx`).

I *did* try to eliminate `docker-compose` as a possible source with my actual problem situation by starting the container with `docker run ` and a lot of `-v` options which resulted in the same `EOF` problem mentioned above.

### Information

The problem seemed to start with the latest `edge` build but I am not sure. As mentioned below the `stable` version does *not* have this problem. The error message varies but besides this the problem is persistent (reproducable).

### Diagnostic logs

This is actually a log from a fresh diagnostics run because I forgot to copy this output from the previous run. Time between diag runs less than 5 minutes.
```
Docker for Mac: version: 18.05.0-ce-mac66 (07d57837a9c0ad813a1042c9dc39bbc5961d031d)
macOS: version 10.13.4 (build: 17E202)
logs: /tmp/0720BB9D-AB80-41C1-94BB-4F5E1A512088/20180604-093612.tar.gz
[OK]     vpnkit
[OK]     virtualization hypervisor
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     disk

$ docker version

Client:
 Version:      18.05.0-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   f150324
 Built:        Wed May  9 22:12:05 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.05.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.10.1
  Git commit:   f150324
  Built:        Wed May  9 22:20:16 2018
  OS/Arch:      linux/amd64
  Experimental: true
```

### Steps to reproduce the behavior

  1. Docker compose file is attached ([docker-compose.txt](https://github.com/docker/for-mac/files/2067287/docker-compose.txt))
  2. A friend tried on the stable version; no troubles at all starting containers. His `docker version` output:
```
$ docker version
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true

$ docker-compose up 
Creating network "xxx_default" with the default driver
Creating xxx_thingie2_1 ... done
Creating xxx_thingie3_1 ... done
Creating xxx_thingie_1  ... done
Attaching to xxx_thingie_1, xxx_thingie2_1, xxx_thingie3_1
xxx_thingie_1 exited with code 0
xxx_thingie2_1 exited with code 0
xxx_thingie3_1 exited with code 0
```

**Comment by steefdebruijn** [comment]
Addendum: I just updated MacOS **and** Docker edge and problem persists:
```
Docker for Mac: version: 18.05.0-ce-mac67 (1fa4e2acfc1a52f79623add2390604515d32297e)
macOS: version 10.13.5 (build: 17F77)
logs: /tmp/0720BB9D-AB80-41C1-94BB-4F5E1A512088/20180608-113707.tar.gz
[OK]     vpnkit
[OK]     virtualization hypervisor
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     disk
```

**Comment by djs55** [comment]
@steefdebruijn Thanks for the bug report and the test case. I've created a fix and it has now been merged into the development branch, so it should be in the next release. If you'd like to try it then there is an experimental build: https://download-stage.docker.com/mac/pr/26054/Docker.dmg Note this build should only be used for testing, not for production :-) and if it offers an auto-update, don't accept or you might end up on a different testing build without the fix.
